api_volontaria/apps/volunteer/tests/test_view_events.py

```diff
@@ -260,3 +263,141 @@ def test_list_events(self):
+        is given as input
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.post(
+            reverse('event-bulk'),
+            data={},
+            format='multipart'
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            content,
+            {'detail': "No file was provided for bulk event creation"}
+        )
+
+    def test_bulk_events_mapping_is_not_valid_json(self):
+        """
+        Ensure bad request with explanation is returned if mapping
+        is not a valid json string
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        invalid_json = '{"detail": "extra bracket at the end "}}'
+        response = self.client.post(
+            reverse('event-bulk'),
+            data={
+                "file": BytesIO(),
+                "mapping": invalid_json
+            },
+            format='multipart'
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(
+            "Mapping should be a dictionary represented in json, errors",
+            content["detail"]
+        )
+
+    def test_bulk_events_mapping_is_not_a_dict(self):
+        """
+        Ensure bad request with explanation is returned if mapping
+        is not a json string representing a dictionary
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.post(
+            reverse('event-bulk'),
+            data={
+                "file": BytesIO(),
+                "mapping": '["list instead of dict"]'
+            },
+            format='multipart'
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            content,
+            {'detail': "Mapping should be a dictionary pairing "
+                       "the csv column (key) to the element key (value)"}
+        )
+
+    @patch("api_volontaria.apps.volunteer.views.add_bulk_from_file")
+    def test_bulk_events_adding_fail(self, add_bulk_from_file):
+        """
+        Ensure bad request with the caught errors is returned
+        if the bulk operation fails
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        error_message = "error message from add_bulk"
+        add_bulk_from_file.side_effect = InvalidBulkUpdate(
+            error_message
+        )
+
+        response = self.client.post(
+            reverse('event-bulk'),
+            data={"file": BytesIO()},
+            format='multipart'
+        )
+
+        content = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(content, {'detail': error_message})
```

Comment:
When it come to error not relative to fields, we go with `non_field_errors` and not with `detail`

`Detail` is used on special case related to permissions or settings, mostly for technician to debug. `non_field_errors` is used for all errors related to data input, as a global form error, when it not come on a specific field.